lib/AST/ASTDumper.cpp

```diff
@@ -520,7 +520,17 @@ namespace {
         OS << " kind=" << getImportKindString(ID->getImportKind());
 
       OS << " '";
-      ID->getImportPath().print(OS);
+      // Check if module aliasing was used for the given imported module; for
+      // example, if '-module-alias Foo=Bar' was passed and this module has
+      // 'import Foo', its corresponding real module name 'Bar' should be printed.
+      ImportPath::Builder builderWithRealModuleName;
+      auto path = ID->getImportPath(/*outRealModuleName=*/&builderWithRealModuleName);
+      if (!builderWithRealModuleName.get().empty()) {
+        // Module aliasing was used, so print its real module name
+        builderWithRealModuleName.get().print(OS);
+      } else {
+        path.print(OS);
+      }
```

Comment:
With my suggestion above, this becomes:

```suggestion
      // Check if module aliasing was used for the given imported module; for
      // example, if '-module-alias Foo=Bar' was passed and this module has
      // 'import Foo', its corresponding real module name 'Bar' should be printed.
      ImportPath::Builder scratch;
      ID->getRealImportPath(scratch).print(OS);
```

I trust you can see why I prefer the separate-method design here. 😃

Reply:
👍